flow/core/kernel/simulation/traci.py

```diff
@@ -62,10 +93,61 @@ def simulation_step(self):
 
     def update(self, reset):
         """See parent class."""
-        pass
+        if reset:
+            self.time = 0
+        else:
+            self.time += self.sim_step
+
+        # Collect the additional data to store in the emission file.
+        if self.emission_path is not None:
+            kv = self.master_kernel.vehicle
+            for veh_id in self.master_kernel.vehicle.get_ids():
+                t = round(self.time, 2)
+
+                # some miscellaneous pre-processing
+                position = kv.get_2d_position(veh_id)
+
+                # Make sure dictionaries corresponding to the vehicle and
+                # time are available.
+                if veh_id not in self.stored_data.keys():
+                    self.stored_data[veh_id] = dict()
+                if t not in self.stored_data[veh_id].keys():
+                    self.stored_data[veh_id][t] = dict()
+
+                # Add the speed, position, and lane data.
+                self.stored_data[veh_id][t].update({
+                    "speed": kv.get_speed(veh_id),
+                    "lane_number": kv.get_lane(veh_id),
+                    "edge_id": kv.get_edge(veh_id),
+                    "relative_position": kv.get_position(veh_id),
+                    "x": position[0],
+                    "y": position[1],
+                    "headway": kv.get_headway(veh_id),
+                    "leader_id": kv.get_leader(veh_id),
+                    "follower_id": kv.get_follower(veh_id),
+                    "leader_rel_speed":
+                        kv.get_speed(kv.get_leader(veh_id))
+                        - kv.get_speed(veh_id),
+                    "target_accel_with_noise_with_failsafe":
+                        kv.get_accel(veh_id, noise=True, failsafe=True),
+                    "target_accel_no_noise_no_failsafe":
+                        kv.get_accel(veh_id, noise=False, failsafe=False),
+                    "target_accel_with_noise_no_failsafe":
+                        kv.get_accel(veh_id, noise=True, failsafe=False),
+                    "target_accel_no_noise_with_failsafe":
+                        kv.get_accel(veh_id, noise=False, failsafe=True),
+                    "realized_accel":
+                        kv.get_realized_accel(veh_id),
+                    "road_grade": kv.get_road_grade(veh_id),
+                    "distance": kv.get_distance(veh_id),
+                })
```

Comment:
It would be nice if we keep the order of all the columns in a way that is compatible with the pipeline. This way new data will be compatible with previous simulation results. The column ordering for the pipeline is `time, id, x, y, speed, headway, leader_id, target_accel_with_noise_with_failsafe, target_accel_no_noise_no_failsafe, target_accel_with_noise_no_failsafe, target_accel_no_noise_with_failsafe, realized_accel, road_grade, edge_id, lane_number, distance, relative_position`.

Reply:
done!